src/raftstore/store/fsm/store.rs

```diff
@@ -93,8 +93,8 @@ pub struct StoreMeta {
     pub pending_votes: RingQueue<RaftMessage>,
     // the regions with pending snapshots.
     pub pending_snapshot_regions: Vec<Region>,
-    // (region_id, version) -> BiLock.
-    pub merge_locks: HashMap<(u64, u64), Option<Arc<AtomicBool>>>,
+    // source_region_id -> (version, BiLock).
+    pub merge_locks: HashMap<u64, (u64, Option<Arc<AtomicBool>>)>,
```

Comment:
seem that we can merge the version and atomic value into
HashMap<u64, Option<Arc<AtomicU64>>>,

Reply:
I don't think so. Version should be always set.